api/converter_utils.go

```diff
@@ -640,7 +738,7 @@ func edIndexToAddress(index uint64, txn sdk.Transaction, shared []sdk.Address) (
 }
 
 func (si *ServerImplementation) assetParamsToAssetQuery(params generated.SearchForAssetsParams) (idb.AssetsQuery, error) {
-	creator, errorArr := decodeAddress(params.Creator, "creator", make([]string, 0))
+	creator, errorArr := decodeAddressToBytes(params.Creator, "creator", make([]string, 0))
 	if len(errorArr) != 0 {
 		return idb.AssetsQuery{}, errors.New(errUnableToParseAddress)
```

Comment:
As an example of what I described earlier, I think this reads much better, without needing to understand the unusual error handling model of `decodeAddress`.

```suggestion
	creator, err := sdk.DecodeAddress(params.Creator)
	if err {
		return idb.AssetsQuery{}, fmt.Errorf("Unable to parse creator address: %v", err)
```

It's worth noting that the existing code supplies the word "creator" to `decodeAddressToBytes` in order to get a nicer error message, and then immediately throws that nice error message away in favor of a generic message.

You can also just use `creator[:]` in the code below, rather than have separate functions to decode to an `sdk.Address`, or to `[]byte`.

Reply:
This change makes a lot of sense:
* removes unnecessary code (there are two functions that pretty much do the same thing as `sdk.DecodeAddress`)
* the logic in those functions is somewhat confusing to the reader
* handling errors as `[]error` is preferrable over `[]string`

Commit https://github.com/algorand/indexer/pull/1638/commits/7f26e0fbc48bfd655fcb0cc1cf6b3fced7e7fc8c adds more context information to the errors being returned. If this is considered a breaking change, I can make it return the same exact old error messages.

I'm not a fan of returning multiple errors, I think it is not worth the extra complexity compared to a fail-fast approach (returning upon the first error encountered). In fact this could introduce bugs, since the control flow keeps advancing with variables that may be in an unknown state (I'd rather stop the control flow immediately upon finding something that doesn't check out).
But there may be a reason why multiple errors are being used that I'm not aware of.

There are several instances in the codebase that are handling multiple errors as `[]string` instead of `[]error`. I didn't change all of these to avoid scope creep, but could do so in a separate PR.